### ISSUE TITLE
write functions to create, manipulate and access specgraph_t struct

### ISSUE THREAD

**Issue by lukasg8** [uchicago-cs/chiventure#1325]
As we will update the speclist_t struct to be an adjacency matrix, we need to write functions to create specgraph_t structs, manipulate its data (especially because game authors may add, remove and edit rooms in their games) and access it.

We would also want to add some sort of visualizing functionality to help game authors see the relationships between their rooms easily.

**Comment by williamhu314** [comment]
We will be implementing and creating tests for new, init, and free functions for specgraphs in #1357 so this issue will focus on other functions that can manipulate and access specgraphs.

**Comment by williamhu314** [comment]
The additional functionalities mentioned above are no longer needed as we have decided to keep both the list of roomspecs and the adjacency matrix static. This is because there is no need to allow the author to adjust the parameters of generation while our generation functions are running-if the author wants to make any adjustments, he/she can just change the information they inputted statically and then rerun the generation functions. The only functions we will need for now are new, init, and free functions for our specgraph struct, which we have implemented in #1357.
